Initial commit of the slapd-base role

dasJ opened the pull request:
This role sets up a basic slapd server with an empty OLC. It's mainly to
remove any distribution-specific stuff from slapd to create a plain and
naked slapd.

dasJ committed:
Initial commit of the slapd-base role

This role sets up a basic slapd server with an empty OLC. It's mainly to
remove any distribution-specific stuff from slapd to create a plain and
naked slapd.

Mr-Pi commented on meta/main.yml:
```diff
@@ -0,0 +1,24 @@
+    - yakkety
+  - name: Debian
+    versions:
+    - wheezy
```
kein jessie?

dasJ replied:
Nein, ollen Scheiß supporte ich nicht

Mr-Pi replied:
wheezy ist old(7), jessie ist 8 ;)

dasJ replied:
Da siehst du mal wie sehr mich beschissene veraltete Distributionen interessieren...... Ich ändere das daheim. Morgen. Mit einem rebase. Erinnere mich bitte, weil ich das nicht werde.

Mr-Pi commented on tasks/olc-setup.yml:
```diff
@@ -0,0 +1,53 @@
+    mode: 0770
+
+- name: Hash OLC password
+  command: "/bin/bash -c '/usr/sbin/slappasswd -n -T /dev/stdin < <(echo -n $password)'"
```
warum so kompliziert?
und keine möglichkeit ein schon gehashtes password zu hinterlegen?

dasJ replied:
Naja, das Passwort soll ja aus dem Vault kommen. Da wollen wir ja nicht das plain Passwort (das wir brauchen um uns später anzumelden) **und** das gehashte Passwort speichern.

Die standard Ansible Hash funktionen gehen nicht, weil slapd da sein eigenes Format hat das etwa so aussieht: `{SSHA}4ICefbyn1VqbvoI3ydYlv5p1XFjxqkl2`

Mr-Pi replied:
Ja das kenne ich, das war aber auch nur irgendwas nomales, zwar nen selten genutzter standard, aber immerhin, ich suche das nochmal raus, wenn gewünscht

dasJ replied:
Was halt nicht mehr gilt wenn man ein anderes Hashingformat konfiguriert (z.B. SHA-2, wo ich mal schauen muss ob Ubuntu das nativ supportet)

Mr-Pi commented on vars/main.yml:
```diff
@@ -0,0 +1,3 @@
+---
+olccreated_flag: "{{ flags_dir }}/.olccreated.flag"
+olc_tmp: /tmp/olc.conf
```
warum in vars und nicht in defaults?

dasJ replied:
Weil es diese Variablen nur gibt, damit ich den String nicht immer runterkopieren muss. Die sollten nicht geändert werden.

Mr-Pi replied:
okay alles klar

Mr-Pi replied:
Willst du für das tmp nicht lieber http://docs.ansible.com/ansible/tempfile_module.html nutzen?

dasJ replied:
Kann ich schon machen. Aber nicht heute. Und vielleicht auch nicht dieses WE

dasJ replied:
Doch nicht: ` In case you need to make them world-accessible you need to use file module.`

dasJ commented:
Pls rereview

dasJ committed:
Do not add potentially unneeded schemas

Schemas will be added later during slapd-config to be more flexible.

dasJ committed:
Add core schema with schema2ldif

This is done instead of hardcoding the include to have cleaner schema
definitions in the OLC.

dasJ commented:
Pls rereview

dasJ commented:
@Mr-Pi Wie siehts aus?

dasJ committed:
Move MDB to /var/lib to comply with FHS
